### PR TITLE
Twitch.tv Window React

### PR DESCRIPTION
**Pull request by lunulae**
Requested in #5380 
I wasn't sure what other livestream services people use, so I only wrote up one for Twitch. I also wasn't sure if I used the right tag for the window detection (just "twitch" rather than "twitch.tv") so please feel free to change it if the full url would work better.

(I haven't made a window reaction before, and I mostly tried to use the Youtube one as a template here.)

### CODE REVIEW

**Comment by multimokia** [comment]
Potential issue I see with this is that there's a potential ambiguity here. (We can't use `twitch.tv` since that's not going to be in the window handle)

But it means it's also going to trigger if I searched up something like `"Why do birds twitch their wings?"`

**Comment by lunulae** [comment]
That's a really good point, and I'm unsure if there's a good solution for it. The only thing I can see that distinguishes between streams in the Twitch window handle is the individual stream titles themselves.

No worries if this can't be used for that reason, I'd rather not make any awkward false positives.

**Comment by multimokia** [comment]
Yeah, sadly with the way windowreacts work, it's based *purely* off the name of the window handle. So for Twitch in google chrome, if I was watching a stream by myself for example, it'd show up as `Multimokia - Twitch - Google Chrome`. Which just doesn't give us enough information, since they're supposed to be generic in order to actually work with programs other than just your browser.

Not really your fault at all, but merely a limitation based on the information we get from the site. If Twitch actually used the `Twitch.tv` bit in the url it'd be fine (even though there'd be some false positives, at least it'd be related to the actual service and can be accounted for in dialogue)

File: Monika After Story/game/zz_windowreacts.rpy [hunk 1]
```diff
@@ -688,3 +688,31 @@ label mas_wrs_reddit:
     if not wrs_success:
         $ mas_unlockFailedWRS('mas_wrs_reddit')
     return
+
+init 5 python:
+    addEvent(
+        Event(
+            persistent._mas_windowreacts_database,
+            eventlabel="mas_wrs_twitch",
+            category=['twitch'],
```
Comment (jmwall24) [inline Monika After Story/game/zz_windowreacts.rpy]:
```suggestion
            category=['-twitch'],
```
I did some testing on this, and using `-twitch` allows it to catch it when you are on any individual streams ('think user - Twitch' for the window name) without it also catching any window that just had `twitch` on it.

**Pull request merged**